src/utils/pipeline.js

```diff
@@ -0,0 +1,64 @@
+const {
+  whitespace, newline, newlineCharacters, beginningOfStringOrNewline
+} = require('./regexes')
+
+const stageItem = `${newline}+${whitespace}*\\- \\[(?<checked> |x)\\] (?<itemName>[^${newlineCharacters}]+)`
+const stageName = `${whitespace}*${newline}#* ?\\*{0,2}(?<stage>[a-z0-9]+[^${newlineCharacters}]*):?\\*{0,2}:?`
+const stageBlock = `${stageName}(${whitespace}*${stageItem})+`
+
+const pipelineHeading = `#* ?\\*{0,2}pipeline:?\\*{0,2}:?`
+const pipelineBlock = `${beginningOfStringOrNewline}${pipelineHeading}${whitespace}*(?<pipeline>(${stageBlock})+)`
```

Comment:
it seems like this makes sure all the checkboxes in the Issue are crossed off before automatically merging... is that correct? And does it do this based on some pipeline text in the PR?

Reply:
yea for example https://github.com/openstax/unified/issues/872 we have a bunch of checkboxes in our issue template. the goal is to trigger auto merge when:
- all the checkboxes in review are checked
- there is a special label on the pr (this may be temporary as i get comfy with the system)
- the pr is not the base of other prs
- status checks pass on the pr